extensions/interactions/LogicProof/static/js/student.js

```diff
@@ -994,12 +994,16 @@ var logicProofStudent = (function() {
   *          }
   */
   var buildProof = function(proofString, questionInstance) {
-    if (proofString.slice(-1) === '\n') {
-      proofString = proofString.slice(0, proofString.length - 1);
-    }
     var lineStrings = proofString.split('\n');
+    // Ignore blank lines at the end

```

Comment:
.trim() would remove spaces at the start of the first line, which I don't want to do.

Lines in the middle are screened for later. This is just a courtesy to clean up accidental newlines at the end.